pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -572,7 +572,7 @@ func fitsResourceQuota(fName kueue.ResourceFlavorReference, rName corev1.Resourc
 
 	cohortUsed := used
 	if cq.Cohort != nil {
-		cohortUsed = cq.Cohort.Usage[fName][rName]
+		cohortUsed = cq.UsedCohortQuota(fName, rName)
 	}
 
 	lack := cohortUsed + val - cohortAvailable
```

Comment:
I'm not convinced that this formula works.

I think you need to calculate `UsedCohortQuota` for `cohortUsed+val` instead of just `cohortUsed`, then substract that from cohortAvailable.

But consider an implementation based on my previous comment.

Reply:
According to our [design](https://github.com/kubernetes-sigs/kueue/pull/1385/files#diff-6f7611f9158b1c0d7bbc28edd666c01b781e7f074243a85e8d3b9ab95e6bf529), when we [accumulate](https://github.com/kubernetes-sigs/kueue/pull/1385/files#diff-bd8411463d27f0b9477a855409c5133c437d8cae368eeff5c2e5957f26c23507R189-R198) `cohort.Usage` in snapshot, we will substract the guaranteedQuota, only count how much quota used in `cohort.RequestableResources`(sum of `LendingLimit`).
And in function `UsedCohortQuota`, we count the `cqUsage` in when the cq is not using all its guaranteedQuota(otherwise count the guaranteedQuota in).